core/common/src/main/java/alluxio/security/authentication/SaslStreamServerDriver.java

```diff
@@ -117,14 +117,10 @@ public void onNext(SaslMessage saslMessage) {
   @Override
   public void onError(Throwable throwable) {
     if (mChannelId != null) {
-      LOG.warn("Closing authenticated channel: {} due to error: {}", mChannelId, throwable);
-      mAuthenticationServer.unregisterChannel(mChannelId);
-    }
-    if (mSaslServerHandler != null) {
-      try {
-        mSaslServerHandler.close();
-      } catch (IOException exc) {
-        LOG.debug("Failed to close SaslServer.", exc);
+      LOG.debug("Closing authenticated channel: {} due to error: {}", mChannelId, throwable);
+      if (!mAuthenticationServer.unregisterChannel(mChannelId)) {
+        // Channel was not registered. Close driver explicitly.
```

Comment:
does this deserve a debug message?

Reply:
Makes sense. added in the unregisterChannel() impl. instead for less duplication.